src/tg/centralized_pca_on_pruning_accuracy.py

```diff
@@ -0,0 +1,154 @@
+import pandas as pd
+import numpy as np
+
+from torch.nn.functional import cross_entropy
+from pytorch_lightning.trainer import Trainer
+from pytorch_lightning.callbacks import EarlyStopping, LearningRateMonitor, ModelCheckpoint
+
+from utils.plink import run_plink
+from nn.models import MLPClassifier
+from configs.phenotype_config import get_accuracy
+from configs.split_config import FOLDS_NUMBER
+from preprocess.pruning import PlinkPruningRunner
+from tg.data_provider import DataProvider
+
+
+LOG_FILE = '/home/genxadmin/centralized-pca-pruning-accuracy.log'
+SPLIT_DIRECTORY = '/mnt/genx-bio-share/TG/data/chip/superpop_split'
+
+
+def get_model(num_classes, num_features, trainer=None):
+    parameters = {
+        'nclass': num_classes,
+        'nfeat': num_features,
+        'optim_params': {
+            'name': 'sgd',
+            'lr': 0.1
+        },
+        'scheduler_params': {
+            'rounds': '10000',
+            'epochs_in_round': 1,
+            'name': 'exponential_lr',
+            'gamma': 0.999
+        },
+        'loss': cross_entropy
+    }
+
+    if not trainer:
+        return MLPClassifier(**parameters)
+    else:
+        parameters['checkpoint_path'] = trainer.checkpoint_callback.best_model_path
+        return MLPClassifier.load_from_checkpoint(**parameters)
+
+
+def get_trainer():
+    return Trainer(
+        max_epochs=10000,
+        callbacks=[
+            EarlyStopping(
+                monitor='val_loss',
+                patience=1000,
+                strict=False,
+                verbose=False,
+                mode='min'
+            ),
+            LearningRateMonitor(logging_interval='step'),
+            ModelCheckpoint(monitor='val_loss', mode='min')
+        ],
+        precision=32,
+        weights_summary='full'
+    )
+
+
+def log(pruning_threshold, variants_number, accuracy_train, accuracy_validation, accuracy_test):
+    with open(LOG_FILE, 'a') as log_file:
```

Comment:
a question from the top of my head: if you're appending the log file, do you clean it before running the whole thing? So that it doesn't contain logs from previous runs

Reply:
Yes, it is re-created on startup.